### PR TITLE
Duplo 4746 : Support for k8s ingress resource.

### PR DESCRIPTION
**Pull request by tahir-duplo**
- `duplocloud_k8_ingress`  resource added.

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-4746 TF provider support for ingress resources](https://app.clickup.com/t/8655600/DUPLO-4746)

File: duplocloud/resource_duplo_k8_ingress.go [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+	tenantID, name, err := parseK8sIngressIdParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[TRACE] resourceK8IngressRead(%s, %s): start", tenantID, name)
+
+	// Get the object from Duplo, detecting a missing object
+	c := m.(*duplosdk.Client)
+	rp, err := c.DuploK8sIngressGet(tenantID, name)
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_k8_ingress.go:166]: I think we should also handle 404 - all v3 methods are supposed to return 404 when the resource does not exist.  we are also updating the backend to do that.

File: duplocloud/resource_duplo_k8_ingress.go [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if rp != nil && rp.Name != "" {
+		err := c.DuploK8sIngressDelete(tenantID, name)
+		if err != nil {
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_k8_ingress.go:250]: I think we should also handle 404 - all v3 methods are supposed to return 404 when the resource does not exist. we are also updating the backend to do that.

File: duplocloud/resource_duplo_k8_ingress.go [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+	// First, set the simple fields.
+	d.Set("tenant_id", tenantId)
+	d.Set("name", duplo.Name)
+	d.Set("ingress_class_name", duplo.IngressClassName)
+
+	// Set LBConfig
+	d.Set("lbconfig", []interface{}{flattenK8sIngressLBConfig(duplo.LbConfig)})
+
+	// Set Rules
+	d.Set("rule", []interface{}{flattenK8sIngressRules(duplo.Rules)})
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_k8_ingress.go:279]: I think this is an extra array wrapping an array.  since the method already returns an array.

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: duplocloud/resource_duplo_k8_ingress.go (hunk 1 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_k8_ingress.go:166]: Done

File: duplocloud/resource_duplo_k8_ingress.go (hunk 2 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_k8_ingress.go:250]: Done

File: duplocloud/resource_duplo_k8_ingress.go (hunk 3 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_k8_ingress.go:279]: Done

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**